NWL/NWL.py

```diff
@@ -451,6 +486,41 @@ def NWL_plot(
     # Define number of source particles
     num_parts = len(coords)
 
-    NWL_mat = count_mat*n_energy*eV2J*SS*J2MJ/num_parts
+    neutron_crossing_mat = count_mat*n_energy*eV2J*SS*J2MJ/num_parts
+
+    # construct array of bin boundaries
+    bin_arr = np.zeros((num_phi+1, num_theta+1, 3))
+
+    for phi_bin, phi in enumerate(phi_bins):
+        for theta_bin, theta in enumerate(theta_bins):
+            
+            if theta < theta_min:
+                theta = theta_min
+            elif theta > theta_max:
+                theta = theta_max
+
+            if phi < phi_min:
+                phi = phi_min
+            elif phi > phi_max:
+                phi = phi_max
+
+            x,y,z = vmec.vmec2xyz(wall_s,theta,phi)
```

Comment:
I think the `source_mesh` code now has a way to generate a list of x,y,z for a list of theta, phi without a loop

Reply:
I wasn't able to find anything like that, I saw a couple spots that used vmec2xyz but they were in loops. Maybe you can point me to the specific bit of code? Sorry if I missed it

Reply:
I guess it's in `invessel` and it's sort of a loop, but embedded in a function so the loop only needs to be written once, and it probably isn't worth relying on it here...

Reply:
That said, my gut says there is a way to do this all with numpy arrays that relies on fewer loops somehow...